app/mattermost.js

```diff
@@ -2,15 +2,38 @@
 // See License.txt for license information.
 
 import React from 'react';
+import {AppState} from 'react-native';
 import {Provider} from 'react-redux';
 
+import {setAppState} from 'service/actions/general';
+import {loadStorage} from 'app/actions/storage';
 import configureStore from 'app/store';
 import Router from 'app/navigation/router';
 import RootLayout from 'app/layouts/root_layout/root_layout_container';
 
 const store = configureStore();
 
 export default class Mattermost extends React.Component {
+    constructor(props) {
+        super(props);
+
+        setAppState(AppState.currentState)(store.dispatch, store.getState);
+        loadStorage()(store.dispatch, store.getState);
+        this.handleAppStateChange = this.handleAppStateChange.bind(this);
+    }
+
+    componentDidMount() {
+        AppState.addEventListener('change', this.handleAppStateChange);
+    }
+
+    componentWillUnmount() {
+        AppState.removeEventListener('change', this.handleAppStateChange);
+    }
+
+    handleAppStateChange(appState) {
+        setAppState(appState)(store.dispatch, store.getState);
```

Comment:
Two things here:
1) I don't think we want to just dump the app state into the store. Since that part of the store will exist in both the webapp and native, we should come up with some sort of interface for it that has the fields that we care about, even if this is just an `isActive` field for now with an action to change that.
2) This should go in the RootLayout. We should avoid talking to the store from outside of Redux.